src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -69,26 +69,42 @@ public partial class AuthenticationResult
             Guid correlationID)
         {
             _authenticationScheme = authenticationScheme ?? throw new ArgumentNullException(nameof(authenticationScheme));
+            string homeAccountId =
+                msalAccessTokenCacheItem?.HomeAccountId ??
+                msalIdTokenCacheItem?.HomeAccountId;
+            string environment = msalAccessTokenCacheItem?.Environment ??
+                msalIdTokenCacheItem?.Environment;
 
-            if (msalAccessTokenCacheItem.HomeAccountId != null)
+            if (homeAccountId != null)
             {
-                string username = msalAccessTokenCacheItem.IsAdfs ? msalIdTokenCacheItem?.IdToken.Upn : msalIdTokenCacheItem?.IdToken?.PreferredUsername;
+                string username = null;
+                if (msalIdTokenCacheItem != null)
+                {
+                    username = msalIdTokenCacheItem.IsAdfs ? 
+                        msalIdTokenCacheItem?.IdToken.Upn : 
+                        msalIdTokenCacheItem?.IdToken?.PreferredUsername;
+                }
+
                 Account = new Account(
-                    msalAccessTokenCacheItem.HomeAccountId,
+                    homeAccountId,
                     username,
-                    msalAccessTokenCacheItem.Environment);
+                    environment);
+            }
+
+            if (msalAccessTokenCacheItem!= null)
+            {
+                AccessToken = authenticationScheme.FormatAccessToken(msalAccessTokenCacheItem);
+                ExpiresOn = msalAccessTokenCacheItem.ExpiresOn;
+                ExtendedExpiresOn = msalAccessTokenCacheItem.ExtendedExpiresOn;
+                Scopes = msalAccessTokenCacheItem.ScopeSet;
+                IsExtendedLifeTimeToken = msalAccessTokenCacheItem.IsExtendedLifeTimeToken;
+                TokenType = msalAccessTokenCacheItem.TokenType;
             }
 
-            AccessToken = authenticationScheme.FormatAccessToken(msalAccessTokenCacheItem);
             UniqueId = msalIdTokenCacheItem?.IdToken?.GetUniqueId();
```

Comment:
>UniqueId [](start = 12, length = 8)

seems like this doesn't fallback to the subject in case the IDT is null? Is the doc wrong on the fallback value?